test/absinthe/schema/rule/input_output_types_correctly_placed_test.exs

```diff
@@ -56,4 +56,74 @@ defmodule Absinthe.Schema.Rule.InputOutputTypesCorrectlyPlacedTest do
       ])
     end
   end
+
+  describe "sdl schema" do
+    test "is enforced with output types on arguments" do
+      assert_schema_error("invalid_output_types_sdl", [
+        %{
+          extra: %{
+            field: :blah,
+            parent: Absinthe.Blueprint.Schema.ObjectTypeDefinition,
+            struct: Absinthe.Blueprint.Schema.InputObjectTypeDefinition,
+            type: :input
+          },
+          locations: [
+            %{
+              file: "test/support/fixtures/dynamic/invalid_output_types_sdl.exs",
+              line: 4
+            }
+          ],
+          phase: Absinthe.Phase.Schema.Validation.InputOutputTypesCorrectlyPlaced
+        },
+        %{
+          extra: %{
+            argument: :invalid_arg,
+            struct: Absinthe.Blueprint.Schema.ObjectTypeDefinition,
+            type: :user
+          },
+          locations: [
+            %{
+              file: "test/support/fixtures/dynamic/invalid_output_types_sdl.exs",
+              line: 4
+            }
+          ],
+          phase: Absinthe.Phase.Schema.Validation.InputOutputTypesCorrectlyPlaced
+        }
+      ])
+    end
+
+    test "is enforced with input types on arguments" do
+      assert_schema_error("invalid_input_types_sdl", [
```

Comment:
This test asserts there are two errors, whilst the macro based schema only has one. This points to an issue with the macro based schema in that input object fields are `FieldDefinition` structs and not `InputValueDefinition` structs like there are in the SDL schema.